screen/wallets/entropyGenerator.js

```diff
@@ -0,0 +1,426 @@
+    setPaths([{ segments: [], color }]);
+  };
+
+  const { colors } = useTheme();
+  const { addWallet, saveToDisk, isAdancedModeEnabled, wallets, isTestModeEnabled } = useContext(BlueStorageContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [walletBaseURI, setWalletBaseURI] = useState('');
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const [label, setLabel] = useState(labelProps || '');
+  const [isAdvancedOptionsEnabled, setIsAdvancedOptionsEnabled] = useState(false);
+  const [isTestMode, setIsTestMode] = useState(false);
+  const [selectedWalletType, setSelectedWalletType] = useState(selectedWalletTypeProps || false);
+  const [backdoorPressed, setBackdoorPressed] = useState(1);
+
+  const [entropyButtonText, setEntropyButtonText] = useState(loc.wallets.add_entropy_provide);
+  const stylesHook = {
+    advancedText: {
+      color: colors.feeText,
+    },
+    label: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+    noPadding: {
+      backgroundColor: colors.elevated,
+    },
+    root: {
+      backgroundColor: colors.elevated,
+      marginBottom: 30,
+    },
+    lndUri: {
+      borderColor: colors.formBorder,
+      borderBottomColor: colors.formBorder,
+      backgroundColor: colors.inputBackgroundColor,
+    },
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem(isTestMode ? AppStorage.TEST_LNDHUB : AppStorage.LNDHUB)
+      .then((url) => setWalletBaseURI(url || (isTestMode ? 'https://tnlndhub.mintlayer.org' : 'https://lndhub.io')))
+      .catch(() => setWalletBaseURI(''));
+    isTestModeEnabled().then(setIsTestMode);
+    isAdancedModeEnabled()
+      .then(setIsAdvancedOptionsEnabled)
+      .finally(() => setIsLoading(false));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isAdvancedOptionsEnabled]);
+
+  const generateEntropyDrawing = async () => {
+    const flatArray = paths.flatMap((path) => path.segments);
+    const numArr = flatArray.flatMap((str) => {
+      // Split the string into parts by spaces
+      const parts = str.split(' ');
+      // Convert the numerical parts into numbers
+      const nums = parts.slice(1).map(parseFloat);
+      // Return an array with the first part and the numbers
+      return nums;
+    });
+    const normalizedPoints = normalize(numArr.map((point) => Math.round(point)));
+    return generateEntropy(normalizedPoints);
+  };
+
+  const createWallet = async () => {
+    setIsLoading(true);
+
+    let w;
+
+    if (selectedWalletType === ButtonSelected.OFFCHAIN) {
+      createLightningWallet(w);
+    } else if (selectedWalletType === ButtonSelected.ONCHAIN) {
+      const entropy = await generateEntropyDrawing();
+      const shuffledEntropy = getNRandomElementsFromArray(entropy, 16);
+      if (entropy.length < 192) {
+        alert('Your entropy is too small. Please draw more lines.');
```

Comment:
```suggestion
        alert('Not enough entropy gathered. Please continue drawing.');
```

Reply:
@owlsua  please update your message as well , so we can be sync